### PR TITLE
Savesystem improvements

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR will improve the save system to an usable state, we can load and save to custom file names that will be included in the final binary, paving the way for proper levels to be made and stored. Finishes OPA-66.

Blocked on #27 so do not merge until #27 is merged!

### CODE REVIEW

File: Assets/Scripts/Core/GameController.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 namespace OperationBlackwell.Core {
 	public class GameController : MonoBehaviour {
+		public static GameController Instance {get; private set;}
```
Comment (lokiy999) [inline Assets/Scripts/Core/GameController.cs:7]: Coding style: add a space before get and after private set; in this file

File: Assets/Scripts/Core/MapEditorSaveSystem.cs [hunk 2]
```diff
@@ -6,9 +6,9 @@
 namespace OperationBlackwell.Core {
 	public static class SaveSystem {
 
-		private const string SAVE_EXTENSION = "txt";
+		private const string SAVE_EXTENSION = "json";
 
-		private static readonly string SAVE_FOLDER = Application.dataPath + "/SavesWorld/";
+		private static readonly string SAVE_FOLDER = Application.streamingAssetsPath + "/SavesWorld/";
```
Comment (lokiy999) [inline Assets/Scripts/Core/MapEditorSaveSystem.cs:11]: Coding style: private variables need to have a trailing "_"

File: Assets/Scripts/Core/Tilemap.cs [hunk 3]
```diff
@@ -75,7 +75,7 @@ public enum NodeSprite {
 			// Holds the amount of cover this tile gives.
 			public bool cover {get; private set;}
 			// Holds if the tile can be walked over.
-			public bool walkable; // {get; protected set;}
+			public bool walkable {get; protected set;}
```
Comment (lokiy999) [inline Assets/Scripts/Core/Tilemap.cs:78]: Coding style: add a space before get and protected set; in this file

File: Assets/Scripts/Core/GameController.cs (hunk 1 above)
Comment (Dennisbonke) [reply Assets/Scripts/Core/GameController.cs:7]: The style this is currently in matches the style used in other files, and is thus correct.

File: Assets/Scripts/Core/MapEditorSaveSystem.cs (hunk 2 above)
Comment (Dennisbonke) [reply Assets/Scripts/Core/MapEditorSaveSystem.cs:11]: For these kinds of constants we don’t do that. However, the isInit variable _does_ need one, so good catch.

File: Assets/Scripts/Core/GameController.cs (hunk 1 above)
Comment (Westlanderz) [reply Assets/Scripts/Core/GameController.cs:7]: Interesting as the first PR where these kind of statements are used is #12 and there it is with extra spaces.

File: Assets/Scripts/Core/GameController.cs [hunk 4]
```diff
@@ -19,17 +19,17 @@ public class GameController : MonoBehaviour {
 		public delegate void MoveEvent(Vector3 position);
 		public event MoveEvent Moved;
 
-		private Grid<Tilemap.Node> grid_;
-		private Tilemap tilemap_;
+		public Grid<Tilemap.Node> grid {get; private set;}
+
 		public GridPathfinding gridPathfinding { get; private set; }
+		public Tilemap tilemap {get; private set;}
```
Comment (Westlanderz) [inline Assets/Scripts/Core/GameController.cs]: Inconsistent with the spacing after `{ get;`

**Review by lokiy999 — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Core/GameController.cs (hunk 4 above)
Comment (Dennisbonke) [reply Assets/Scripts/Core/GameController.cs]: Already fixed.

**Pull request merged**